filters/auth/jwt_metrics_test.go

```diff
@@ -3,155 +3,200 @@ package auth_test
 				}},
 			},
-			response: &http.Response{StatusCode: http.StatusOK},
+			status: http.StatusOK,
 			expected: map[string]int64{
-				"GET.foo_test.200.invalid-issuer": 1,
+				"jwtMetrics.custom.GET.foo_test.200.invalid-issuer": 1,
 			},
 		},
 		{
-			name: "no invalid-issuer for empty issuers",
-			def:  `jwtMetrics()`,
+			name:    "no invalid-issuer for empty issuers",
+			filters: `jwtMetrics()`,
 			request: &http.Request{Method: "GET", Host: "foo.test",
 				Header: http.Header{"Authorization": []string{
 					"Bearer header." + marshalBase64JSON(t, map[string]any{"iss": "baz"}) + ".signature",
 				}},
 			},
-			response: &http.Response{StatusCode: http.StatusOK},
+			status:   http.StatusOK,
 			expected: map[string]int64{},
 		},
 		{
-			name: "no invalid-issuer when matches first",
-			def:  `jwtMetrics("{issuers: [foo, bar]}")`,
+			name:    "no invalid-issuer when matches first",
+			filters: `jwtMetrics("{issuers: [foo, bar]}")`,
 			request: &http.Request{Method: "GET", Host: "foo.test",
 				Header: http.Header{"Authorization": []string{
 					"Bearer header." + marshalBase64JSON(t, map[string]any{"iss": "foo"}) + ".signature",
 				}},
 			},
-			response: &http.Response{StatusCode: http.StatusOK},
+			status:   http.StatusOK,
 			expected: map[string]int64{},
 		},
 		{
-			name: "no invalid-issuer when matches second",
-			def:  `jwtMetrics("{issuers: [foo, bar]}")`,
+			name:    "no invalid-issuer when matches second",
+			filters: `jwtMetrics("{issuers: [foo, bar]}")`,
 			request: &http.Request{Method: "GET", Host: "foo.test",
 				Header: http.Header{"Authorization": []string{
 					"Bearer header." + marshalBase64JSON(t, map[string]any{"iss": "bar"}) + ".signature",
 				}},
 			},
-			response: &http.Response{StatusCode: http.StatusOK},
+			status:   http.StatusOK,
+			expected: map[string]int64{},
+		},
+		{
+			name:    "missing-token without opt-out",
+			filters: `jwtMetrics("{issuers: [foo, bar], optOutAnnotations: [oauth.disabled]}")`,
+			request: &http.Request{Method: "GET", Host: "foo.test"},
+			status:  http.StatusOK,
+			expected: map[string]int64{
+				"jwtMetrics.custom.GET.foo_test.200.missing-token": 1,
+			},
+		},
+		{
+			name: "no metrics when opted-out",
+			filters: `
+				annotate("oauth.disabled", "this endpoint is public") ->
+				jwtMetrics("{issuers: [foo, bar], optOutAnnotations: [oauth.disabled, jwtMetrics.disabled]}")
+			`,
+			request:  &http.Request{Method: "GET", Host: "foo.test"},
+			status:   http.StatusOK,
+			expected: map[string]int64{},
+		},
+		{
+			name: "no metrics when opted-out by alternative annotation",
+			filters: `
+				annotate("jwtMetrics.disabled", "skip jwt metrics collection") ->
+				jwtMetrics("{issuers: [foo, bar], optOutAnnotations: [oauth.disabled, jwtMetrics.disabled]}")
+			`,
+			request:  &http.Request{Method: "GET", Host: "foo.test"},
+			status:   http.StatusOK,
```

Comment:
Maybe add a test when annotation from `annotate` does not match with annotation from `jwtMetrics`?

Reply:
Thanks, added.